src/types.rs

```diff
@@ -118,6 +118,7 @@ fn long_signature(name: &str, params: &[ParamType]) -> Hash {
 }
 
 #[derive(Default, BorshDeserialize, BorshSerialize, Clone)]
+#[cfg_attr(test, derive(serde::Deserialize, serde::Serialize))]
```

Comment:
The same thing here.

Reply:
I think `test` is what we want here because the `serde` (de)serialization is only used in our own tests that are gated by `#[cfg(test)]`. If another crate were to depend on this one then I think it would be bad to expose the `serde` (de)serialization since we don't want others to be using it. So the fact that the crate can never be compiled with the `test` flag by other dependencies I think is a feature here.